File: 1716-leetcode-bank.py
def sumn(n):
    return (n * (n-1))//2
class Solution:
    def totalMoney(self, n: int) -> int:
        total = 0
        for i in range(7):
            count = (n+i)//7
            total += (7-i) * count
            total += sumn(count)
        return total

File: 1716-leetcode-bank.rs
#[inline]
fn sumn(n:i32) -> i32 {
    (n * (n-1))/2
}
impl Solution {
    pub fn total_money(n: i32) -> i32 {
        (0..7).map(|i| {
            let count = (n+i)/7;
            (7-i) * count + sumn(count)
        }).sum()
    }
}
